test/test_irods_http_api.py

```diff
@@ -2113,6 +2113,54 @@ def test_server_reports_error_when_op_is_not_supported(self):
     def test_return_error_on_missing_parameters(self):
         pass
 
+    def test_server_handles_binary_data_encoded_as_multipart_form_data_correctly(self):
+        headers = {'Authorization': 'Bearer ' + self.rodsuser_bearer_token}
+        data_object = f'/{self.zone_name}/home/{self.rodsuser_username}/embedded_crlfs.txt'
+
+        try:
+            # Create a data object using multipart/form-data encoding.
+            # The body of the request will contain an embedded CRLF. The CRLF is what triggers
+            # the bug in the original parser. This would cause the data to be truncated.
```

Comment:
Maybe worth referencing #273 either here or in the test name? Don't have to, but could be helpful in addition to the explanation here.

Reply:
Appended statement referencing the issue number.